### PR TITLE
Align numerical data

### PR DESCRIPTION
**Pull request by Befkadu1**
fix: Lundalogik/crm-feature#2123
## Review:
- [x] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [x] Commits have the correct *type* for the changes made
- [x] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [x] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/table/table.types.ts [hunk 1]
```diff
@@ -28,6 +29,11 @@ export interface Column<T extends object = any> {
      * A component used to render inside the column header
      */
     headerComponent?: TableComponentDefinition;
+
+    /**
+     * Sets the horizontal text alignment for the column (left|center|right)
+     */
+    hozAlign?: Tabulator.ColumnDefinitionAlign;
```
Comment (adrianschmidt) [inline src/components/table/table.types.ts:36]: Kind of nitpicky, but I don't really like the shortening `hozAlign`. (I understand it comes from Tabulator.) I think I'd like it to be either `horizontalAlign` or, if we want to go short, just `hAlign`… 🤔

@jgroth Thoughts?

File: src/components/table/columns.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export class ColumnDefinitionFactory {
         const definition: Tabulator.ColumnDefinition = {
             title: column.title,
             field: column.field,
+            hozAlign: column.hozAlign,
```
Comment (adrianschmidt) [inline src/components/table/columns.ts:22]: I think the first two commits can be squashed together.

File: src/components/table/table.types.ts (hunk 1 above)
Comment (Befkadu1) [reply src/components/table/table.types.ts:36]: I agree with you. I was about to write it like that at the beginning. When I updated the examples I had to use `hozAlign` . For example `{ title: 'Height', field: 'height', hozAlign: 'right' },`. I thought of consistency that is the reason I used same name everywhere

Comment (adrianschmidt) [reply src/components/table/table.types.ts:36]: Hmm… ok. I'll have a quick look at it, and if I can't figure out how to make it work, we'll just go with `hozAlign` 🙂

Comment (jgroth) [reply src/components/table/table.types.ts:36]: Yeah I don't like `hozAlign` either. I cannot see why we would not be able to use `horizontalAlign` :thinking:

Comment (jgroth) [reply src/components/table/table.types.ts:36]: Also, we should not expose types from Tabulator like this. If they make changes to it we might make breaking changes in our own API without realizing it.

Comment (adrianschmidt) [reply src/components/table/table.types.ts]: I pushed fixups for making it `hAlign`, but it's easy to go from there to `horizontalAlign`.

Comment (adrianschmidt) [reply src/components/table/table.types.ts]: I changed it to `horizontalAlign`.

Comment (Befkadu1) [reply src/components/table/table.types.ts]: > Also, we should not expose types from Tabulator like this. If they make changes to it we might make breaking changes in our own API without realizing it.

What do you think if we have our columnDefinationAlign then?
Like this `type ColumnDefinitionAlign = 'left' | 'center' | 'right';`

Comment (jgroth) [reply src/components/table/table.types.ts]: I think we can just inline it for now. Or do you think it's useful to have a separate type for it?

Comment (adrianschmidt) [reply src/components/table/table.types.ts]: Yeah, I tried with this:

```ts
    /**
     * Sets the horizontal text alignment for the column
     */
    horizontalAlign?: 'left' | 'center' | 'right';
```

and it looks like this:

image

👌

Comment (Befkadu1) [reply src/components/table/table.types.ts]: We can squash those changes and merge.

**Review by adrianschmidt — APPROVED** [review]
I changed the scope of the commit from `column` to `table`, as that will make more sense to someone reading the changelog. I also updated the commit message slightly.

Good work 👍

**Pull request merged**